src/components/Modals/CreateSingleItemModal/EditThumbnailStep/EditThumbnailStep.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from 'react'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { PreviewMessageType, PreviewOptions, sendMessage } from '@dcl/schemas'
+import { ModalNavigation, WearablePreview, Row, Button, Icon, Loader } from 'decentraland-ui'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+import BuilderIcon from 'components/Icon'
+import { ControlOptionAction, Props, State } from './EditThumbnailStep.types'
+import './EditThumbnailStep.css'
+
+const DEFAULT_ZOOM = 3
+
+export default class EditThumbnailStep extends React.PureComponent<Props, State> {
+  previewRef = React.createRef<WearablePreview>()
+  state: State = {
+    blob: this.props.blob,
+    previewController: this.props.wearablePreviewController,
+    hasBeenUpdated: false,
+    frame: 0,
+    isPlaying: false,
+    disableAutoRotate: false
+  }
+
+  clearPlayingInterval = () => {
+    const { playingIntervalId } = this.state
+    if (playingIntervalId) {
+      clearInterval(playingIntervalId)
+      this.setState({ playingIntervalId: undefined })
+    }
+  }
+
+  componentDidUpdate(_prevProps: Props, prevState: State) {
+    const { isPlaying } = this.state
+    if (prevState.isPlaying && !isPlaying) {
+      this.clearPlayingInterval()
+    }
+  }
+
+  componentWillUnmount() {
+    const { playingIntervalId } = this.state
+    if (playingIntervalId) {
+      clearInterval(playingIntervalId)
+    }
+  }
+
+  restart = () => {
+    const { length } = this.state
+    this.clearPlayingInterval()
+    this.setState({ isPlaying: true, frame: 0 }, () => length && this.trackFrame(length))
+  }
+
+  trackFrame = (length: number, currentFrame?: number) => {
+    const { isPlaying, playingIntervalId } = this.state
+    if (isPlaying && playingIntervalId) {
+      return
+    }
+
+    let counter = currentFrame || 0
+    let max = length * 100
```

Comment:
What's length defined in? What if we navigate the frames by a fixed adjustable percentage?

Reply:
It's defined in seconds. Would you mind putting an example of how to navigate it with a fixed adjustable %?